test/TestConfig/scripts/testKitGen/makeGenTool/mkgen.pl

```diff
@@ -93,41 +91,38 @@ sub runmkgen {
 		print "Getting modes data from modes services...\n";
 	}
 
-	#initialize counting hash targetCount and disabledTable
+	# initialize counting hash targetCount
 	$targetCount{"all"} = 0;
-	foreach my $eachLevel (sort @{$allLevels}) {
-		if (!defined $targetCount{$eachLevel}) {
-			$targetCount{$eachLevel} = 0;
-			$disabledTable{$eachLevel} = [];
-		}
-		foreach my $eachGroup (sort @{$allGroups}) {
-			if (!defined $targetCount{$eachGroup}) {
-				$targetCount{$eachGroup} = 0;
-				$disabledTable{$eachGroup} = [];
-			}
-			my $lgKey = $eachLevel . '.' . $eachGroup;
-			if (!defined $targetCount{$lgKey}) {
-				$targetCount{$lgKey} = 0;
-				$disabledTable{$lgKey} = [];
+	my @allDisHead = ('', 'disabled.', 'echo.disabled.');
+	foreach my $eachDisHead (@allDisHead) { 
+		$targetCount{$eachDisHead . "normal.all"} = 0;
```

Comment:
When we count the targets, I think we don't need `normal.all`. We just need `all` and it includes all the disabled targets as well, then the number will match the total number in the result summary. The same logic should be applied to other targets, such as sanity, sanity.functional, etc.